ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/IdeaFrame.kt

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.uitests.fixtures
+
+import com.intellij.remoterobot.RemoteRobot
+import com.intellij.remoterobot.data.RemoteComponent
+import com.intellij.remoterobot.fixtures.CommonContainerFixture
+import com.intellij.remoterobot.fixtures.ContainerFixture
+import com.intellij.remoterobot.fixtures.DefaultXpath
+import com.intellij.remoterobot.fixtures.FixtureName
+import com.intellij.remoterobot.search.locators.byXpath
+import com.intellij.remoterobot.stepsProcessing.step
+import com.intellij.remoterobot.utils.waitFor
+import java.time.Duration
+
+fun RemoteRobot.idea(function: IdeaFrame.() -> Unit) {
+    find<IdeaFrame>().apply(function)
+}
+
+@FixtureName("Idea frame")
+@DefaultXpath("IdeFrameImpl type", "//div[@class='IdeFrameImpl']")
+class IdeaFrame(remoteRobot: RemoteRobot, remoteComponent: RemoteComponent) : CommonContainerFixture(remoteRobot, remoteComponent) {
+    val projectViewTree
+        get() = find<ContainerFixture>(byXpath("ProjectViewTree", "//div[@class='ProjectViewTree']"))
+
+    val projectName
+        get() = step("Get project name") { return@step callJs<String>("component.getProject().getName()") }
+
+    @JvmOverloads
+    fun dumbAware(timeout: Duration = Duration.ofMinutes(5), function: () -> Unit) {
+        step("Wait for smart mode") {
+            waitFor(duration = timeout, interval = Duration.ofSeconds(5)) {
+                runCatching { isDumbMode().not() }.getOrDefault(false)
```

Comment:
do we have to worry about the IDE entering dumb mode randomly?

Reply:
Not sure, I did not write this but took it from the WIP example and I have not used this code myself yet.